### ISSUE TITLE
yosys `flatten` ERROR with `chip`

### ISSUE THREAD

**Issue by phyzhenli** [PrincetonUniversity/OPDB#2]
Hi~

I use yosys to parse `X_1__Y_1__NETWORK_2dmesh__L1D_8192_4__L1I_16384_4__L15_8192_4__L2_65536_4/chip.pickle.v`.

After `read_verilog` and `hierarchy`, the `flatten` command throws an ERROR: `Cell port lsu.dcache.dcd_fuse_repair_en is driving constant bits \dcd_fuse_repair_en <= \dcache.dcd_fuse_repair_en`.

Thanks.

**Comment by phyzhenli** [comment]
Similar errors were found in `L1I_16384_4__L1D_8192_4/sparc_core.pickle.v`, `L1I_16384_4/sparc_ifu.pickle.v`, `L1D_8192_4/sparc_lsu.pickle.v`, and `NETWORK_2dmesh__L1D_8192_4__L1I_16384_4__L15_8192_4__L2_65536_4/tile.pickle.v`.

**Comment by EdwardMotivo** [comment]
I had a similar issue; I think I encountered variants of the issue in 2 or 3 blocks.  My fix was to move the assign statements for _dcd_fuse_repair_value_ & _dcd_fuse_repair_en_ from a higher level module to a lower-level module where they are tagged as outputs.  For example, in the current version of the sparc_core block the _lsu_ module has the following two assign statements:

```
module lsu (...)
...
assign dcd_fuse_repair_en = 2'b0;
assign dcd_fuse_repair_value = 2'b0;
...
endmodule
```

I removed those two assign statements, and inserted replacement statements in the lower-level _bw_r_dcd_ module just below where the _dcd_fuse_repair_X_ variables are tagged as outputs:

```
module bw_r_dcd (...)
...
output [7:0]    dcd_fuse_repair_value;  //data out for redundancy register
output [1:0]    dcd_fuse_repair_en;     //enable bits out
assign dcd_fuse_repair_value = 8'b0;
assign dcd_fuse_repair_en = 2'b0;
...
endmodule
```

I also changed the bit length of the dcd_fuse_repair_value assignment to match with the variable size, although maybe that causes other problems.  Regardless, my change fixed the error.  

I had to make the same change for the _icd_fuse_repair_value_ and _icd_fuse_repair_en_ variables in the sparc_core block, and I think in another block or two, to get things working.
